packages/next-intl/test/middleware/middleware.test.tsx

```diff
@@ -301,6 +301,68 @@ describe('prefix-based routing', () => {
       expect(MockedNextResponse.next).toHaveBeenCalled();
     });
   });
+
+  describe('localePrefix: never', () => {
+    const middleware = createMockMiddleware({
+      defaultLocale: 'en',
+      locales: ['en', 'de'],
+      localePrefix: 'never'
+    });
+
+    it('rewrites requests for the default locale', () => {
+      middleware(createMockRequest('/'));
+      expect(MockedNextResponse.next).not.toHaveBeenCalled();
+      expect(MockedNextResponse.redirect).not.toHaveBeenCalled();
+      expect(MockedNextResponse.rewrite.mock.calls[0][0].toString()).toBe(
+        'http://localhost:3000/en'
+      );
+    });
+
+    it('rewrites requests for other locales', () => {
+      middleware(createMockRequest('/', 'de'));
+      expect(MockedNextResponse.next).not.toHaveBeenCalled();
+      expect(MockedNextResponse.redirect).not.toHaveBeenCalled();
+      expect(MockedNextResponse.rewrite.mock.calls[0][0].toString()).toBe(
+        'http://localhost:3000/de'
+      );
+    });
+
+    it('rewrites requests for the default locale at a nested path', () => {
+      middleware(createMockRequest('/list'));
+      expect(MockedNextResponse.next).not.toHaveBeenCalled();
+      expect(MockedNextResponse.redirect).not.toHaveBeenCalled();
+      expect(MockedNextResponse.rewrite.mock.calls[0][0].toString()).toBe(
+        'http://localhost:3000/en/list'
+      );
+    });
+
+    it('rewrites requests for other locales at a nested path', () => {
+      middleware(createMockRequest('/list', 'de'));
```

Comment:
Can you also include some tests where the `accept-language` header and the cookie value don't match while no locale is in the pathname? The cookie value should have precedence in this case.

Reply:
Of course.